demos/common/python/models/segmentation.py

```diff
@@ -84,3 +84,15 @@ def postprocess(self, outputs, meta):
 
         result = cv2.resize(result, (input_image_width, input_image_height), 0, 0, interpolation=cv2.INTER_NEAREST)
         return result
+
+
+class SalientObjectDetectionModel(SegmentationModel):
+
+    def postprocess(self, outputs, meta):
+        input_image_height = meta['original_shape'][0]
+        input_image_width = meta['original_shape'][1]
+        mask = outputs[self.out_blob_name].squeeze()
+        mask = 1/(1 + np.exp(-mask))
+        mask = mask.astype(np.uint8)
+        result = cv2.resize(mask, (input_image_width, input_image_height), 0, 0, interpolation=cv2.INTER_NEAREST)
```

Comment:
Correct me if I'm wrong, but doesn't the demo essentially assign random colors to different values in the segmentation result? Won't that result in nonsensical-looking output with a saliency map? Seems like you would get a much more helpful output if the colors were proportional to the values (in fact, you could just treat the map as a grayscale image).

Reply:
Previously, I binorize mask before visualization (0 for background, 1 for attention object), so it is not a problem with random colors, but approach with grayscale is better, so I changed visualization process

Reply:
there is several variants how we can visualazie results
1) full grayscale mask
![salient_full](https://user-images.githubusercontent.com/29454499/112830232-ebc74300-909a-11eb-9aa1-204650a1bf28.png)
2) combination of frame and map like in segmentation demo
![salient_half](https://user-images.githubusercontent.com/29454499/112830331-0a2d3e80-909b-11eb-8c57-feb7ec4640ef.png)
3) use saliency map as mask for frame (looks like background removal)
![salient_overlay](https://user-images.githubusercontent.com/29454499/112830451-38128300-909b-11eb-9ff1-57b51a34ac94.png)

Original picture as reference
![1803151818-00000003](https://user-images.githubusercontent.com/29454499/112830584-6abc7b80-909b-11eb-916d-6ddaee78dda2.jpg)

Which way is preferable?

Reply:
#2 is aligned with what segmentation demo does right now.

Reply:
I would say \#2 as well.

Reply:
ok, then I stay this variant as it is implemented as current solution